pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2175,12 +2177,10 @@ func buildListener(opts buildListenerOpts) *listener.Listener {
 	return listener
 }
 
-// appendListenerFallthroughRoute adds a filter that will match all traffic and direct to the
-// PassthroughCluster. This should be appended as the final filter or it will mask the others.
-// This allows external https traffic, even when port the port (usually 443) is in use by another service.
-func appendListenerFallthroughRoute(l *listener.Listener, opts *buildListenerOpts,
-	node *model.Proxy, currentListenerEntry *outboundListenerEntry) {
-	wildcardMatch := &listener.FilterChainMatch{}
+// Create pass through filters for the listener assuming all the other filter chains are ready.
+// The match member of pass through filter chain depends on the existing non-passthrough filter chain.
+// TODO: Calculate the filter chian match to replace the wildcard and replace appendListenerFallthroughRoute.
```

Comment:
TODO(lambdai)

Reply:
is this one outdated, as you developed? Seems like you already finished what this describes?

Reply:
Add my id.
This is not the end of a perfect pass through.
There are actually lots of holes: we expect match-none filter chain traffic ends up hitting pass through filter chain.
The reality is that some traffic actually cannot find any filter chain and TCP connection will be closed.

Currently we will suggest user to improve the VS to cover more holes. Not sure if we should pay super complicated config for these corner case traffic.